mg/target_test.go

```diff
@@ -0,0 +1,64 @@
+package mg
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestTarget(t *testing.T) {
+	t.Parallel()
+	dirs := []string{"testdata/src_dir", "testdata/target_dir"}
+	for _, d := range dirs {
+		os.MkdirAll(d, 0777)
+		time.Sleep(time.Second)
+	}
+	files := []string{
+		"testdata/src_dir/some_file",
+		"testdata/src_dir/some_",
+		"testdata/src_file",
+		"testdata/target_dir/some_file",
+		"testdata/target_file",
+	}
+	for _, v := range files {
+		err := ioutil.WriteFile(v, []byte(v), 0600)
+		if err != nil {
+			t.Fatal(err)
+		}
+		time.Sleep(time.Second)
```

Comment:
you can make this sleep 10ms.... no need to make this test take 5 seconds.  10ms will still be plenty to spread out the file creation times.